### DISCUSSION TITLE
Local Ory/Kratos and SDK

### DISCUSSION THREAD

**Discussion by valentinge** [ory/kratos#2189] [Q&A] [answered]
Good day!

I apologize if I missed something in the documentation; I'm pretty new to this domain.

I'm trying to integrate ory/kratos' functionality in our gateway. I managed to run the ory/kratos image locally and test its endpoints with Postman. My next step would be to use your SDK, but it seems that I need an URL similar to this one to create an instance `https://playground.projects.oryapis.com/api/kratos/public`. I'm using [kratos-selfservice-ui-react-native](https://github.com/ory/kratos-selfservice-ui-react-native) as a reference.

Can I use the SDK with the public base_url from the Kratos configuration? Is it possible to use the SDK with locally hosted Kratos?

I apologize again! I'm surely missing a big part of the whole picture. Any guidance is welcome!.

Kind regards,
Valentin

**Comment by gen1us2k** [comment, marked as answer]
Hello. We use openapi generator for our SDKs. You can find all of them [here](https://www.ory.sh/kratos/docs/sdk). Also, [HTTP API documentation](https://www.ory.sh/kratos/docs/reference/api) can be useful for your case.  Public API runs on 4433 port and can be exposed to the internet using ngrok or similar tools.

You can also check out our [quickstart example](https://www.ory.sh/kratos/docs/quickstart) to better understand how it works

**Reply by valentinge** [reply]
@gen1us2k, thank you for your swift response!

You mentioned that the HTTP API could be useful in my case. Would that be the default go-to if we're making server-to-server calls only, or we can take advantage of the SDK as well?

Tanks again!

Ps: We have already started implementing our abstraction of your REST API, and it seems to be working smoothly.

**Reply by aeneasr** [reply]
You can also try the `kratos-client` which is fully compatible with the open source. We will soon make the regular client fully compatible with both open source and self-hosted :)

**Reply by valentinge** [reply]
Gentlemen, thank you for your attention!

Is the `kratos-client` the mentioned regular client? Also, as we tried to adapt your REST API, I ran into an issue finding the 
**oryAccessToken** required for some of the identity operations.

The self-serving endpoints are working wonderfully!

Ps: I found that the REST endpoints that require authorization by the mentioned token work perfectly fine when I use my test account for the Ory Cloud. Also, my manager sent me the following post: https://news.ycombinator.com/item?id=29330048.

**Reply by vinckr** [reply]
Hello @valentinge 
Please correct me if wrong @aeneasr,
but I believe for self-hosted Ory Kratos you currently should use the `kratos-client` package but this is in the process of being migrated to the general ory `client` package: https://www.npmjs.com/package/@ory/client. This is most likely also the cause of the `oryAccessToken` issue here.
There is still some kinks to work out before the migration is done, we will update the documentation then accordingly. 
PS: should be resolved soon 🙂

**Reply by valentinge** [reply]
Gentlemen, I don't want to waste much more of your time. We're considering switching from the default SQLite to MySQL so we can do whatever else we need on our own with a complementary microservice.

I'm still interested in the **oryAccessToken**. Maybe there's a way to generate such tokens other than the Ory Cloud's dashboard option. I'm probably missing some clues. If you can shed some light on that, that would be the last piece of crucial info.

Thank you again!

**Reply by tdipadova3rd** [reply]
@vinckr @aeneasr I'm wondering the same thing about `oryAccessToken`s. I've searched all over the documentation, but I can't find any reference to how to generate these when self-hosting. Is this something we need Ory Hydra for in addition to Kratos? Thanks in advance!

**Reply by vinckr** [reply]
Hello @valentinge, @tdipadova3rd
The `oryAccessToken` is a feature that is provided through Ory Network (https://console.ory.sh/) and not part of Ory Kratos. 
When self-hosting Ory Kratos you have to make sure to protect admin endpoints, see also the [deploy guide](https://www.ory.sh/docs/kratos/guides/deploy-kratos-example) for this.
You do not need Ory Hydra for this, rather a different service altogether, that Ory is not (yet) offering. There are plans for such a service at Ory, but it is in planning stage at the moment.
